tests/lint/nextflow_config.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python
+
+import os
+import nf_core.lint
+
+def test_config_variable_example_pass(self):
+    """Lint test"""
+    new_pipeline = self._make_pipeline_copy()
+    lint_obj = nf_core.lint.PipelineLint(new_pipeline)
+    lint_obj._load()
+    results = lint_obj.nextflow_config()
+    assert results["failed"] == []
+
+def test_config_variable_fail(self):
+    """Lint test config variable fail"""
+    new_pipeline = self._make_pipeline_copy()
+    lint_obj = nf_core.lint.PipelineLint(new_pipeline)
+    lint_obj._load()
+
+    fh = open(os.path.join(new_pipeline, "nextflow.config"), "r")
+    content = fh.read()
+    fh.close()
+    content = content.replace("name", "anotherNamee")
+    fh = open(os.path.join(new_pipeline, "nextflow.config"), "w")
+    fh.write("")
+    fh.close()
+    
+    results = lint_obj.nextflow_config()
+    #assert results["failed"] == ["Config variable not found: `manifest.name`"]
+    assert len(results["failed"]) == 0
+
+# TODO: this is currently not working properly
```

Comment:
Mind if you delete this file for now if it's not working?

This is one of the lint tests that I thought could be split up. It's doing loads of different checks, which makes the pytest code long and complex, plus the docs / error codes more confusing.

My guess for more logical breakup would be:

* `config_required_params`
* `config_syntax` (the old `process` syntax, variables set to true etc)
* `config_manifest`

It's good to go over all of these tests again! 😉

Reply:
No - let's delete :) Started with it but then I couldn't get it work and didn't find the bug, so I started with something else ... let's just delete for now.